core/src/data_sources/data_source.rs

```diff
@@ -296,43 +296,30 @@ impl DataSource {
         &self.config
     }
 
-    fn qdrant_collection(&self) -> String {
+    pub fn qdrant_collection(&self) -> String {
         format!("ds_{}", self.internal_id)
     }
 
-    pub async fn setup(
+    pub async fn update_config(
```

Comment:
The only thing we ever want to update is `qdrant_config` no ? Would it be better to only expose that ?

Reply:
I would have kept the `sore` fn flexible, but only expose what you can actually change in `update_config` as this is the interface for the rest of the app.
It's "clear" that you can do "unsupported" things by calling `store`, but to me `data_source` functions shouldn't leave things in an inconsistent state